danielMunteanu/src/main/java/Generics/GenericLibrary.java

```diff
@@ -0,0 +1,35 @@
+package Generics;
+
+import java.util.List;
+
+public class GenericLibrary<T extends LibraryItem> {
+    private List<T> items;
+
+    public List<T> getItems() {
+        return items;
+    }
+
+    public void setItems(List<T> items) {
+        this.items = items;
+    }
+
+    public void printByName(List<T> libraryItems, String itemName) {
+
+        for (int i = 0; i < items.size(); i++) {
+            if (itemName.contains(items.get(i).getName())) {
+                System.out.println(items.get(i));
+            }
+        }
+        System.out.println("No more" + " items with name " + itemName + " where found.");
+    }
+
+    public void printByName(String itemName) {
```

Comment:
What if I call this method with "Harry"? I'm expecting it will print "Harry Potter" book.

Reply:
fixed,now works as intended from comment